src/sqlfluff/dialects/dialect_exasol.py

```diff
@@ -203,7 +203,10 @@
         Ref("UDFParameterDotSyntaxSegment"),
     ),
     FunctionScriptTerminatorSegment=NamedParser(
-        "function_script_terminator", CodeSegment, type="function_script_terminator"
+        "function_script_terminator",
+        SymbolSegment,
+        name="function_script_terminator",
+        type="function_script_terminator",
     ),
```

Comment:
@WittierDinosaur - one of the main changes I made here was setting the `name` kwarg.